modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/actions/jms/client/Send.java

```diff
@@ -77,95 +79,106 @@
-                message = new MapCarbonMessage();
-                MapCarbonMessage mapCarbonMessage = (MapCarbonMessage) message;
-                if (ballerinaMessageDataSource instanceof BMap) {
-                    BMap mapData = (BMap) ballerinaMessageDataSource;
-                    for (Object o : mapData.keySet()) {
-                        BValue key = (BValue) o;
-                        BValue value = mapData.get(key);
-                        mapCarbonMessage.setValue(key.stringValue(), value.stringValue());
-                    }
-
-                    message = mapCarbonMessage;
-                }
-            } else if (!(message instanceof MapCarbonMessage)) {
-                throw new BallerinaException(
-                        "If the message type is MapMessage, either set MapData property or pass a received" +
-                        " jms map message", context);
-            }
-            propertyMap.put(JMSConstants.JMS_MESSAGE_TYPE, JMSConstants.MAP_MESSAGE_TYPE);
-        } else {
-            propertyMap.put(JMSConstants.JMS_MESSAGE_TYPE, JMSConstants.GENERIC_MESSAGE_TYPE);
+        // Create message content according to the message type.
+        String messageType = message.getHeader(Constants.CONTENT_TYPE);
+        if (messageType == null) {
+            messageType = Constants.OCTET_STREAM;
         }
-        propertyMap.put(JMSConstants.DESTINATION_PARAM_NAME, getStringArgument(context, 0));
+
+        message.setHeader(JMSConstants.JMS_MESSAGE_TYPE, messageType);
+        switch (messageType) {
+            case Constants.TEXT_PLAIN:
+            case Constants.APPLICATION_JSON:
+            case Constants.APPLICATION_XML:
+                message = getTextCarbonMessage(bMessage, messageType, context);
+                break;
+            case Constants.APPLICATION_FORM:
+                message = getMapCarbonMessage(bMessage, context);
+                break;
+            case Constants.OCTET_STREAM:
+            default:
+                message = getBlobCarbonMessage(bMessage, context);
+        }
+
+        propertyMap.put(JMSConstants.DESTINATION_PARAM_NAME, destination);
         try {
             if (log.isDebugEnabled()) {
-                log.debug("Sending " + messageType + " to " +
-                          propertyMap.get(JMSConstants.DESTINATION_PARAM_NAME));
+                log.debug("Sending " + messageType + " to " + propertyMap.get(JMSConstants.DESTINATION_PARAM_NAME));
             }
-            //Getting the sender instance and sending the message.
+
             BallerinaConnectorManager.getInstance().getClientConnector("jms")
                                      .send(message, null, propertyMap);
         } catch (ClientConnectorException e) {
-            throw new BallerinaException("Exception occurred while sending message.", e, context);
+            throw new BallerinaException("Failed to send message. " + e.getMessage(), e, context);
```

Comment:
Do we need to add the e.getMessage(), as we are sending the exception element itself.